src/liquid/components/ld-badge/ld-badge.tsx

```diff
@@ -0,0 +1,77 @@
+import { Component, Element, h, Host, Prop, State, Watch } from '@stencil/core'
+import { HTMLStencilElement } from '@stencil/core/internal'
+import { getClassNames } from 'src/liquid/utils/getClassNames'
+
+/**
+ * @virtualProp ref - reference to component
+ * @virtualProp {string | number} key - for tracking the node's identity when working with lists
+ */
+@Component({
+  assetsDirs: ['assets'],
+  tag: 'ld-badge',
+  styleUrl: 'ld-badge.css',
+  shadow: true,
+})
+export class LdBadge {
+  @Element() el: HTMLStencilElement
+
+  /** Defines badge custom color */
+  @Prop() brandColor?: boolean
+
+  /** Icon name. */
+  @Prop() icon?: string
+
+  /** The size of the badge, translated in rem */
+  @Prop() size?: 'sm' | 'lg'
+
+  @State() hasCustomIcon = false
+  @State() hasText = false
+
+  @Watch('size')
+  private updateIconSize() {
+    this.el.querySelectorAll('ld-icon').forEach((icon) => {
+      if (this.size !== undefined) {
+        icon.setAttribute('size', this.size)
+      } else {
+        icon.removeAttribute('size')
+      }
+    })
+    this.el.querySelectorAll('.ld-icon').forEach((icon) => {
+      if (this.size === 'sm') {
+        icon.classList.remove('ld-icon--lg')
+        icon.classList.add('ld-icon--sm')
+      } else if (this.size === 'lg') {
+        icon.classList.remove('ld-icon--sm')
+        icon.classList.add('ld-icon--lg')
+      } else {
+        icon.classList.remove('ld-icon--sm', 'ld-icon--lg')
+      }
+    })
+  }
+
+  componentWillLoad() {
+    this.hasCustomIcon = !!this.el.querySelector('[slot="icon"]')
+    this.hasText = !!this.el.textContent.trim()
+    this.updateIconSize()
+  }
+
+  render() {
+    const cl = getClassNames([
+      'ld-badge',
+      this.size && `ld-badge--${this.size}`,
+      this.hasText && 'ld-badge--with-text',
+      this.brandColor && `ld-badge--brand-color`,
+    ])
+
+    return (
+      <Host class={cl}>
+        {(this.icon || this.hasCustomIcon) && (
+          <ld-icon class="ld-badge__icon" part="icon" name={this.icon}>
+            <slot name="icon"></slot>
+          </ld-icon>
+        )}
```

Comment:
Okay, now I got it. I think developers may expect the possibility to just do

```html
<ld-badge>
  <ld-icon name="checkmark" slot="icon" />
  done
</ld-badge>
```

And then wonder why this would look crappy. Also it might be a good thing that the developer can decide whether he wants to place the icon on the left or on the right side of a possible text node. But if you leave it like this, I understand that you (of course can) neglect my previous comments.